os/board/imxrt1050-evk/src/imxrt_bringup.c

```diff
@@ -95,20 +95,22 @@
  * Private Functions
  ****************************************************************************/
 
-#if defined(CONFIG_I2C_DRIVER)
+#if defined(CONFIG_I2C)
 static void imxrt_i2c_register(int bus)
 {
 	FAR struct i2c_master_s *i2c;
 	int ret;
+	char path[16];
 
-	i2c = imxrt_i2cbus_initialize(bus);
+	snprintf(path, sizeof(path), "/dev/i2c-%d", bus - 1);
+	i2c = up_i2cinitialize(bus);
 	if (i2c == NULL) {
-		serr("ERROR: Failed to get I2C%d interface\n", bus);
+		syslog("ERROR: Failed to get I2C%d interface\n", bus - 1);
 	} else {
-		ret = i2c_register(i2c, bus);
+		ret = i2c_uioregister(path, i2c);
```

Comment:
Also it is very dangerous approach because you trying to use both of driver model & iotbus at the same time, I mean same bus being used both of driver & iotbus
You "HAVE TO" use one of them, not use together.
Because There should be collision.
maybe i2c / gpio will be safe, but spi, i2s, pwm is not safe because they take more time & usually attached DMA.
Also if gpio is attached DMA, then it also dangerous.
Because none of your iotbus checking bus's state, please assume that bus is already running but you request in the middle of operation. (CPU doesn't care after request send/receive to dmac, it just realize it when it interrupted.)